secboot/secboot_tpm.go

```diff
@@ -308,53 +309,104 @@ func SealKey(key EncryptionKey, params *SealKeyParams) error {
 		return fmt.Errorf("TPM device is not enabled")
 	}
 
+	pcrProfile, err := buildPCRProtectionProfile(numModels, params.ModelParams)
+	if err != nil {
+		return err
+	}
+
+	// Provision the TPM as late as possible
+	if err := tpmProvision(tpm, params.TPMLockoutAuthFile); err != nil {
+		return err
+	}
+
+	// Seal key to the TPM
+	creationParams := sb.KeyCreationParams{
+		PCRProfile: pcrProfile,
+		PINHandle:  pinHandle,
+	}
+	if err := sbSealKeyToTPM(tpm, key[:], params.KeyFile, params.TPMPolicyUpdateDataFile, &creationParams); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
if we're not wrapping the error with additional context message you can simplify to:
```suggestion
	return sbSealKeyToTPM(tpm, key[:], params.KeyFile, params.TPMPolicyUpdateDataFile, &creationParams)
```
and in ResealKey() as well.

Reply:
We were initially wrapping the error, and at some point we removed the extra message to avoid repetition but the code layout wasn't changed. Updating it.